tests/common/plugins/sanity_check/checks.py

```diff
@@ -557,3 +560,125 @@ def _check():
+        for line in stdout.split('\n'):
+            line = line.strip()
+            if len(line) > 33:
+                filename = line[33:].strip()
+                rw_files[filename] = line[:32] #md5sum
+        logger.info("Read %d rw files from dut %s" % (len(rw_files), duthost.sonichost.hostname))
+
+        return results
+
+    def _read_configs():
+        results = {}
+        for duthost in duthosts:
+            config = _read_config_by_dut(duthost)
+            if config:
+              results[duthost.sonichost.hostname] = config
+        return results
+
+    def _do_check(allowlist, filenames, hostname):
+        default_test_allowlist = []
+        conflicts = []
+        allowlist_all = default_allowlist + allowlist
+        for item in allowlist_all:
+            pattern = '^{0}$'.format(item)
+            match = False
+            for filename in filenames:
+                if re.match(pattern, filename):
+                    match = True
+            if not match:
+                logger.error('Unexpected change file found: %s' % filename)
+                conflicts.append(filename)
+
+        return conflicts
+
+    def _pre_check():
+        configs = _read_configs()
+        cache.write(module_cache_config, module, configs)
+
+    def _post_check():
+        check_results = []
+        old_configs = cache.read(module_cache_config, module)
+        if not old_configs:
+            old_configs = {}
+        new_configs = _read_configs()
+        for hostname in new_configs:
+            new_config = new_configs[hostname]
+            new_files = new_config['rw']
+            allowlist = new_config['allowlist']
+            old_config = old_configs.get(hostname, {})
+            old_files = old_config.get('rw', {})
+            change_files = {}
+            for filename in new_files:
+                if filename not in old_files or old_files[filename] != new_files[filename]:
+                    change_files[filename] = hostname
+
+            # Check if the file change is expected
+            check_result = {"failed": False, "check_item": "secureboot", "host": hostname}
+            conflicts = _do_check(allowlist, change_files, hostname)
+            if conflicts:
+                return conflicts
+                check_result["failed"] = True
+                reason = 'Unexpected change files: %s in %s' % (','.join(conflicts), hostname)
+                check_result["failed_reason"] = reason
+            check_results.append(check_result)
+
+        return check_results
+
+    def _check(*args, **kwargs):
+        check_results = []
+        stage = kwargs.get('stage', None)
+
+        if stage == 'stage_pre_test':
+            _pre_check()
+        elif stage == 'stage_post_test':
+            check_results = _post_check()
+        if not check_results:
+            check_result = {"failed": False, "check_item": "secureboot"}
+            check_results.append(check_result)
+
+        return check_results
```

Comment:
The check_results would be an empty list if all secureboot related checks passed. Just checking the final sanity check results, there is no indication that secureboot sanity check has been performed. Is it better for the final secureboot check results to have some successful results (`failed=False`) content in the list?

Reply:
For the pre-test, we do nothing, so only add a fake successful check result.
For the post-test, we will check if the file change is expected, will add result per dut.

---
In reply to: [594070383](https://github.com/Azure/sonic-mgmt/pull/3139#discussion_r594070383) [](ancestors = 594070383)